OIIInspector/image_manager.py

```diff
@@ -0,0 +1,198 @@
+    _grpc_init_wait_time = 5
+    _base_container_name = "OIIInspector_running_image"
+    _image_is_running = False
+    _container_platform = None
+
+    def __init__(self, image_address):
+        """
+        Initialization of the ImageManager
+
+        :param image_address:
+        :type image_address:
+        """
+        self._get_container_platform()
+        self._image_address = None
+        self._port = self._grpc_start_port
+        self._rpc_proc = None
+        self._image_address = image_address
+
+    def __enter__(self):
+        self.start_image()
+        return self
+
+    def __exit__(self, exc_type, exc_value, exc_tb):
+        self.close_image_manager()
+
+    def start_image(self):
+        """
+        Method that starts desired image
+        """
+        # pull needs to be done separately, because command run (which is capable of pull), would be terminated by timer
+        self._pull_image()
+        self._port, self._rpc_proc = self._serve_index_registry()
+        self._image_is_running = True
+
+    def get_local_address_of_image(self) -> str:
+        """
+        Method for getting local address of the running image
+
+        :return: Returns local address where image is running in format localhost:PORT
+        :rtype: str
+        """
+        if self._image_is_running is False:
+            raise (OIIInspectorError("Image is not running"))
+        return f"localhost:{self._port}"
+
+    def close_image_manager(self):
+        """
+        Close and remove running image
+        """
+        self._stop_image_registry()
+        self._rpc_proc.kill()
+        self._remove_image_registry()
+
+    def _get_container_platform(self):
+        """
+        Method searching for available container platform
+        """
+        docker_terminal_result = run_cmd("command -v docker")
+        podman_terminal_result = run_cmd("command -v podman")
+        if "/podman" in podman_terminal_result:
+            self._container_platform = "podman"
+        elif "/docker" in docker_terminal_result:
+            self._container_platform = "docker"
+        else:
+            raise OIIInspectorError("Docker or Podman is needed to be installed on the platform")
+
+    def _remove_old_image_if_exists(self):
+        """
+        Check if some image is running(from previous runs) and if yes, it will stop running image and remove it
+        """
+
+        self._stop_image_registry(tolerate_err=True)
+        self._remove_image_registry(tolerate_err=True)
+
+    def _stop_image_registry(self, tolerate_err=False):
+        """
+        Method that will stop running image with available container platform
+
+        :param tolerate_err: Parameter for set up handling of the exceptions
+        :type tolerate_err: bool
+        """
+        run_cmd(f"{self._container_platform} stop {self._base_container_name}_{self._port}", tolerate_err=tolerate_err)
+
+    def _remove_image_registry(self, tolerate_err=False):
```

Comment:
Same here

Reply:
What about start_image, pull_image and maybe get_local_address_of_image too? Should not be them renamed too, as in other two cases?

Reply:
Also the image manager, should it be container manager then?

Reply:
@querti Could you check the names image/container now? If I understand it right, container is box for running image, to which we are given the image-address. So I renamed few functions in image_manager.py and also in oii_client.py due to mentioned fact.

Reply:
The terminology seems right